enumerable.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    i = 0
+    array2 = []
+    while i < size
+      array2 << self[i] if yield(self[i])
+      i += 1
+    end
+    array2
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
* When no block or argument is given returns true when none of the collection members are false or nil.
```
Failure/Error: expect(true_array.my_all?).to be true_array.all?
     
     LocalJumpError:
       no block given (yield)

```
* When a class is passed as an argument returns true if all of the collection is a member of such class.
```
Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true if all of the collection matches the Regex
```
Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
```
Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```